### PR TITLE
Add query parameters to the url

### PR DESCRIPTION
**Pull request by marko-bekhta**
Fixes quarkusio/quarkusio.github.io#1917

### CODE REVIEW

File: src/main/resources/web/app/qs-form.ts [hunk 1]
```diff
@@ -56,11 +56,27 @@ export class QsForm extends LitElement {
     }
   })
   private _formData: any;
+  private _queryData: any;
```
Comment (yrodiere) [inline src/main/resources/web/app/qs-form.ts:59]: Seems to me that `_backendData`/`_browserData` would be a better description of what these variables do, wouldn't it?

**Review by yrodiere — APPROVED** [review]
Thanks! I didn't test, but I suppose you did, and this looks good to me. I'll let @ia3andy chime in if he wants to have a look.

File: src/main/resources/web/app/qs-form.ts (hunk 1 above)
Comment (marko-bekhta) [reply src/main/resources/web/app/qs-form.ts]: Probably 😃 

btw the URL format is http://localhost:4000/guides/#categories=data&q=orm

Comment (yrodiere) [reply src/main/resources/web/app/qs-form.ts]: What do I know though, maybe `#` makes more sense for client side parameters... I'm clueless.

Comment (marko-bekhta) [reply src/main/resources/web/app/qs-form.ts]: with `#` a page is not forced to be refreshed. If we'd want to go with the `?` then we probably can try
```
window.history.pushState("", "", `?${(new URLSearchParams(this._browserData)).toString()}`)
```
instead, this should also not force a refresh. Let's maybe see if others have any opinions on the URL format. 😃

**Comment by yrodiere** [comment]
> Thanks! I didn't test, but I suppose you did, and this looks good to me. I'll let @ia3andy chime in if he wants to have a look.

Silent minority won't get a vote :) Merging.

**Pull request merged**